
syntax = "proto3";

package fb;
import "google/protobuf/timestamp.proto";

service ChatServer {
	
	// List all chat rooms
	rpc List(Request) returns (ListResponse) {}
	 
	// Join a specific chat room
	rpc Join(Request) returns (Response) {}
	
	rpc Leave(Request) returns (Response) {}
	
	rpc SignIn(Request) returns (Response) {}
	
	//Chat stream over RPC
	rpc Chat(stream Message) returns (stream Message) {}
	
}

message Request {
	string username = 1;
	repeated string params = 2;
}

message Response {
	string text = 1;
}

message Message {
	string id = 1;
	string username = 2;
	string text = 3;
	google.protobuf.Timestamp timestamp = 4;
}

message ListResponse {
	repeated string complete_room_list = 1;
	repeated string joined_room_list = 2;
}